aerofiles/igc/reader.py

```diff
@@ -5,5 +5,144 @@ class Reader:
     see http://carrier.csi.cam.ac.uk/forsterlewis/soaring/igc_file_format/igc_format_2008.html
     """
 
-    def __init__(self, fp=None):
-        self.fp = fp
+    def __init__(self):
+        pass
+
+    def read(self, fp):
+        logger_id, fix_records, task, dgps_records, event_records, satellite_records, security_records, header,\
+            fix_record_extensions, k_record_extensions, k_records, comment_records = (None,) * 12
+
+        for line in fp:
+
+            if line.startswith('A'):
+                logger_id = self.decode_a_record(line)
+            elif line.startswith('B'):
+                if fix_records is None:
+                    fix_records = []
+                fix_records.append(self.decode_b_record(line))
+            elif line.startswith('C'):
+                if task is None:
+                    task = self.decode_c_record(line, first_line=True)
+                else:
+                    array_key, value = self.decode_c_record(line)
+                    task[array_key].append(value)
+            elif line.startswith('D'):
+                if dgps_records is None:
+                    dgps_records = []
+                dgps_records.append(self.decode_d_record(line))
+            elif line.startswith('E'):
+                if event_records is None:
+                    event_records = []
+                event_records.append(self.decode_e_record(line))
+            elif line.startswith('F'):
+                if satellite_records is None:
+                    satellite_records = []
+                satellite_records.append(self.decode_f_record(line))
+            elif line.startswith('G'):
+                if security_records is None:
+                    security_records = []
+                security_records.append(self.decode_g_record(line))
+            elif line.startswith('H'):
+                if header is None:
+                    header = self.decode_h_record(line, first_line=True)
+                else:
+                    key, value = self.decode_h_record(line)
+                    header[key] = value
+            elif line.startswith('I'):
+                fix_record_extensions = self.decode_i_record(line)
+            elif line.startswith('J'):
+                k_record_extensions = self.decode_j_record(line)
+            elif line.startswith('K'):
+                if k_records is None:
+                    k_records = []
+                k_records.append(self.decode_k_record(line))
+            elif line.startswith('L'):
+                if comment_records is None:
+                    comment_records = []
```

Comment:
Ok, will change that.

Something else i'm puzzling with is the best way to deal with the h_records. The format is as follows (spaces added for clarity):
`H F PLT Pilotincharge: John Doe`
 
- The second byte denotes the data source and can contain F(flight recorder) or O(official observer) according to [FAI specification](http://www.fai.org/component/phocadownload/category/?download=11005:igc-fr-spec-with-al4a-2016-4-10)(top of page 24).
- [Here is an IGC file](http://archive.soaringspot.com/contest/013/1323/flights/2477/2597322753.igc) which uses the letter P on the second byte for the copilot:
`H P CM2 Crew2:`. This makes me wonder what other letters are possible.

Now the question becomes: should the source of the header information be included after reading it in?

- It will make the data structure of the h-records a bit more complex.
- but it is more true to the original file

Without data source information if will be something like:

```python
header = {
'pilot': 'John Doe',
'second_pilot': 'John Two',
}
```

Wit data source information if will be something like:

```python
header = {
'pilot': {'data_source': 'F',
          'content': 'John Doe'},
'second_pilot': {'data_source': 'P',
                 'content': 'John Two'}
}
```

It might also be possible to return multiple 'header' variables: `header_f`, `header_p`, `header_o`.